### ISSUE TITLE
Convert logprobs (sometimes positive) to regular probabilities?

### ISSUE THREAD

**Issue by anatol-grabowski** [abb128/april-asr#18]
Hey. How does one convert logprobs to regular probabilities? I've tried `np.exp(token.logprob)` expecting them to be always negative. But sometimes they are positive. Need some explanation on how to handle those.

**Comment by anatol-grabowski** [comment]
Realized that it is impossible to get probability from just one of the logits.
Used a variation of [this code](https://github.com/abb128/LiveCaptions/blob/main/src/line-gen.c#L262) to get an estimate for the probability.

**Comment by abb128** [comment]
I think it may be necessary to add a softmax layer to the output to achieve this. I'll look into it when I get around to training another model
